File: Frontend/src/app/Services/theatre.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpHeaders} from "@angular/common/http";
import {Theatre} from "../Models/theatre.model";

@Injectable({
  providedIn: 'root'
})
export class TheatreService {

  httpOptions = {
    headers :new HttpHeaders({
      'Content-Type':'application/json'
    })
  }

  constructor(private http: HttpClient) { }

  addTheatre(theatre: any) {
    return this.http.post("http://localhost:8090/theatre/add", theatre, this.httpOptions);
  }

  getAllTheatre() {
    return this.http.get<Theatre[]>("http://localhost:8090/theatre");
  }

  getTheatreById(locationid: any, movieid: any) {
    console.log(locationid + movieid);
    return this.http.get<Theatre[]>(`http://localhost:8090/getTheatre/${locationid}/${movieid}`);
  }
}

File: Backend/src/main/java/com/example/backend/Service/Theatre/TheatreService.java
package com.example.backend.Service.Theatre;

import com.example.backend.Model.Theatre;

import java.util.List;

public interface TheatreService {
    public List<Theatre> listAllTheatre();
    public void addTheatre(Theatre theatre);
  
    public List<Theatre> findTheatreByMovieandLocation(int locationId,int movieid);
}
